platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/maps/MapLibreMap.java

```diff
@@ -996,9 +999,35 @@ void notifyStyleLoaded() {
   // Annotations
   //
 
-  public void addSymbol(@NonNull KSymbol symbol) {
-    symbol.attach(this, nextId++);
-    annotationContainer.add(symbol);
+  /* To make the types of annotations more discoverable, we have implemented the same method
+   * four times for all of the subtypes of Annotation.
+   */
+
+  public void addAnnotation(@NonNull Symbol annotation) {
```

Comment:
Nice!

Could you add some doc comments to the new classes here, it will look pretty in the documentation.

Reply:
@louwers Documentation comments are now added.

I also un-deprecated the `removeAnnotations()` method, while rewriting the `clear()` doc comment to refer to `removeAnnotations()`. This decision was made because `clear()` could be understood to also remove other things, like map listeners, and hence `removeAnnotations()` is more explicit about what it actually does.

Reply:
Great.